src/app/api/v1/user/email/route.ts

```diff
@@ -0,0 +1,105 @@
+import AppConstants from "../../../../../appConstants";
+
+import {
+  getSubscriberByEmail,
+  deleteResolutionsWithEmail,
+} from "../../../../../db/tables/subscribers";
+import {
+  addSubscriberUnverifiedEmailHash,
+  removeOneSecondaryEmail,
+  getEmailById,
+} from "../../../../../db/tables/emailAddresses.js";
+
+import { sendVerificationEmail } from "../../../utils/email";
+
+import { validateEmailAddress } from "../../../../../utils/emailAddress";
+import { getL10n } from "../../../../functions/server/l10n";
+import { initEmail } from "../../../../../utils/email";
+
+interface EmailAddRequest {
+  email: string;
+}
+
+export async function POST(req: NextRequest) {
+  const token = await getToken({ req });
+  const l10n = getL10n();
+
+  if (token) {
+    try {
+      const body: EmailAddRequest = await req.json();
+      const subscriber = await getSubscriberByEmail(token.email);
+      const emailCount = 1 + (subscriber.email_addresses?.length ?? 0); // primary + verified + unverified emails
+      const validatedEmail = validateEmailAddress(body.email);
+
+      if (validatedEmail === null) {
+        return NextResponse.json({
+          success: false,
+          status: 400,
+          message: l10n.getString("user-add-invalid-email"),
+        });
+      }
+
+      if (emailCount >= AppConstants.MAX_NUM_ADDRESSES) {
+        return NextResponse.json({
+          success: false,
+          status: 400,
+          message: l10n.getString("user-add-too-many-emails"),
+        });
+      }
+
+      // checkForDuplicateEmail
+      const emailLowerCase = validatedEmail.email.toLowerCase();
+      if (emailLowerCase === subscriber.primary_email.toLowerCase()) {
+        return NextResponse.json({
+          success: false,
+          status: 400,
+          message: l10n.getString("user-add-duplicate-email"),
+        });
+      }
+
+      for (const secondaryEmail of subscriber.email_addresses) {
+        if (emailLowerCase === secondaryEmail.email.toLowerCase()) {
+          return NextResponse.json({
+            success: false,
+            status: 400,
+            message: l10n.getString("user-add-duplicate-email"),
+          });
+        }
+      }
+
+      const unverifiedSubscriber = await addSubscriberUnverifiedEmailHash(
+        subscriber,
+        validatedEmail.email
+      );
+
+      await initEmail();
+      await sendVerificationEmail(subscriber, unverifiedSubscriber.id, l10n);
+
+      return NextResponse.json({
+        success: true,
+        newEmailCount: emailCount + 1,
+        message: "Sent the verification email",
+      });
+    } catch (e) {
+      if (e.message === "error-email-validation-pending") {
+        return NextResponse.json(
+          {
+            success: false,
+            message: "Verification email recently sent, try again later",
+          },
+          { status: 429 }
+        );
+      }
+      return NextResponse.json({ success: false }, { status: 500 });
```

Comment:
Q: Does this need a `message`?

Reply:
I don't think so, since we don't know what the cause of this error is - I don't think we want to expose internal eror messages to the outside world.